secret/secrets_manager_vault.go

```diff
@@ -20,20 +22,36 @@ func NewBasicSecretsManager(c SecretsManagerClient) *BasicSecretsManager {
 
 // CreateSecret creates a new secret.
 func (m *BasicSecretsManager) CreateSecret(ctx context.Context, s NamedSecret) (id string, err error) {
```

Comment:
You can call `s.Validate()` to check that the secret input is valid (this might require you to merge main depending on which commit you branched off).

Reply:
Will do once the other PR is merged in with main branch.